services/workflows-service/src/workflow/workflow.controller.internal.ts

```diff
@@ -42,16 +47,15 @@ export class WorkflowControllerInternal {
   }
 
   @common.Get()
-  @swagger.ApiOkResponse({ type: [WorkflowDefinitionModel] })
-  @swagger.ApiForbiddenResponse()
-  @ApiNestedQuery(WorkflowDefinitionFindManyArgs)
-  async listWorkflowDefinitions(
-    @UserData() userInfo: UserInfo,
-    @common.Req() request: Request,
-  ): Promise<WorkflowDefinition[]> {
-    const args = plainToClass(WorkflowDefinitionFindManyArgs, request.query);
+  @swagger.ApiOkResponse()
+  @swagger.ApiNotFoundResponse({ type: errors.NotFoundException })
+  @swagger.ApiForbiddenResponse({ type: errors.ForbiddenException })
+  @ApiNestedQuery(FindWorkflowsListDto)
+  @UsePipes(new ZodValidationPipe(FindWorkflowsListSchema))
+  async list(@common.Query() { filterId }: FindWorkflowsListDto) {
```

Comment:
Since this controller handles both workflow definition and workflow runtime data renaming `list` to include what is being returned would be helpful.